kube/src/client/mod.rs

```diff
@@ -109,8 +113,7 @@ impl Client {
             http::header::SEC_WEBSOCKET_VERSION,
             HeaderValue::from_static("13"),
         );
-        // TODO generate random key and verify like tungstenite
-        let key = "kube";
```

Comment:
:man_facepalming:

Reply:
hey, it seemed to work :-)
..which i guess begs the question, why? does the apiserver just not care?

Reply:
I guess Kubernetes server is not strictly following RFC6455.

> If the server, while reading the handshake, finds that the client did not send a handshake that matches the description below (note that as per [RFC2616], the order of the header fields is not important), including but not limited to any violations of the ABNF grammar specified for the components of the handshake, the server MUST stop processing the client's handshake and return an HTTP response with an appropriate error code (such as 400 Bad Request).
>
> ...
> 5. A |Sec-WebSocket-Key| header field with a base64-encoded (see Section 4 of [RFC4648]) value that, when decoded, is 16 bytes in length.
>
> https://tools.ietf.org/html/rfc6455#page-21

I remember adding this key because I got `400 Bad Request`, so I guess it doesn't care as long as it exists.

Reply:
hah. well, good thing to add. they'll probably get it right at one point and break us without warning if we are out of spec.